src/fs/fs.c

```diff
@@ -40,6 +40,76 @@ LIST_HEAD(list, cfs_fs_node) fs_list = LIST_HEAD_INITIALIZER();
 /* global endpoint operations for cortxfs*/
 static const struct cfs_endpoint_ops *g_e_ops;
 
+int fs_node_init(struct cfs_fs_node *fs_node, struct namespace *ns,
+		 size_t ns_size)
+{
+	int rc = 0;
+
+	str256_t *fs_name = NULL;
+	ns_get_name(ns, &fs_name);
+
+	fs_node->cfs_fs.ns = calloc(1, ns_size);
+	if (fs_node->cfs_fs.ns == NULL) {
+		goto ns_alloc_fail;
+	}
+
+	memcpy(fs_node->cfs_fs.ns, ns, ns_size);
```

Comment:
Any reason for copying this namespace `ns` to `fs_node->cfs_fs.ns`?
We could have avoided allocating space on line number 51 and just directly written following way.
`fs_node->cfs_fs.ns = ns;` Any thoughts on this?

If you are not agree on above point then `ns` on the caller side needed to be freed which is not done as of now inside `cfs_fs_create` in success case on `out` label because you have duplicated the ns instance here and it should be freed on the caller side(`ns` was allocated by `ns_create`)